src/ishikawa_backend.erl

```diff
@@ -101,40 +105,104 @@ init([]) ->
     %% Generate local version vector.
     VClock = vclock:fresh(),
 
+    %% Generate local stable version vector.
+    SVV = vclock:fresh(),
+
+    %% Generate local recent timestamp matrix.
+    RTM = mclock:fresh(),
+
+    %% Generate local to be delivered messages queue.
+    ToBeDeliveredQueue = [],
+
+    %% Generate local to be acknowledged messages queue.
+    ToBeAckQueue = [],
+
     %% Add membership callback.
     ?PEER_SERVICE:add_sup_callback(fun ?MODULE:update/1),
 
     %% Add initial members.
     {ok, Members} = ?PEER_SERVICE:members(),
     lager:info("Initial membership: ~p", [Members]),
 
-    {ok, #state{actor=Actor, vv=VClock, members=Members}}.
+    {ok, #state{actor=Actor, vv=VClock, members=Members, svv=SVV, rtm=RTM, to_be_delivered_queue=ToBeDeliveredQueue, to_be_ack_queue=ToBeAckQueue}}.
 
 %% @private
 -spec handle_call(term(), {pid(), term()}, #state{}) ->
     {reply, term(), #state{}}.
 handle_call({tcbcast, Message}, _From, #state{actor=Actor,
                                               vv=VClock0,
-                                              members=Members} = State) ->
+                                              members=Members,
+                                              to_be_ack_queue=ToBeAckQueue} = State) ->
     %% First, increment the vector.
     VClock = vclock:increment(Actor, VClock0),
 
     %% Generate message.
-    Message = {tcbcast, Actor, encode(Message), VClock},
+    Message1 = {tcbcast, Actor, encode(Message), VClock},
 
     %% Transmit to membership.
-    [send(Message, Peer) || Peer <- Members],
+    [send(Message1, Peer) || Peer <- Members],
 
-    {reply, ok, State#state{vv=VClock}};
-handle_call({tcbdeliver, _Message, _Timestamp}, _From, State) ->
-    %% TODO: Implement me.
-    {reply, ok, State};
+    %% Add members to the queue of not ack messages.
+    ToBeAckQueue1 = ToBeAckQueue ++ [{{Actor, VClock}, Members}],
```

Comment:
If for some reason the message is dropped, these need to be retried, right?  Otherwise, it will remain unacknowledged forever.